spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraPredicateRules.scala

```diff
@@ -0,0 +1,17 @@
+package org.apache.spark.sql.cassandra
+
+import com.datastax.spark.connector.cql.TableDef
+import org.apache.spark.sql.sources.Filter
+
+case class AnalyzedPredicates(
+  handledByCassandra: Set[Filter],
+  handledBySpark: Set[Filter] ){
+  override def toString(): String = {
+    s"""C* Filters: [${handledByCassandra.mkString(", ")}]
+       |Spark Filters [${handledBySpark.mkString(", ")}]""".stripMargin
+  }
+}
+
+trait CassandraPredicateRules{
+  def applyRules(predicates: AnalyzedPredicates, tableDef: TableDef): AnalyzedPredicates
```

Comment:
maybe just `apply`? woudl be `someCassandraPredicateRule.apply(...)`, or just `someCassandraPredicateRules(...)`